src/jobs/blockchain/blockchain.service.ts

```
@@ -128,4 +159,29 @@ export class BlockchainService {
+  async getTokenBalance(address: string, token: string): Promise<number> {
+    const tokenMeta = SUPPORTED_TOKENS[token];
+    if (!tokenMeta) throw new Error(`Unsupported token: ${token}`);
+    if (token === 'ETH') {
+      // TODO: Implement actual ETH balance check via Starknet provider
+      // For now, mock a balance
+      return 10; // Mock: 10 ETH
+    }
+    if (!tokenMeta.contractAddress)
+      throw new Error('Token contract address missing');
+    // ERC20 balanceOf
+    const contract = this.getContract(tokenMeta.contractAddress, ERC20_ABI);
```

ESLint

[error] 'contract' is assigned a value but never used.

(@typescript-eslint/no-unused-vars)